### PR TITLE
style: simplify code

### PR DESCRIPTION
**Pull request by dristpunk**

### CODE REVIEW

File: solidity/contracts/ConnextVestingWallet.sol [hunk 1]
```diff
@@ -16,184 +15,91 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
   }
 
-  /**
-   * @inheritdoc Ownable2Step
-   * @dev Override needed by linearization
-   */
-  function _transferOwnership(address _newOwner) internal virtual override(Ownable2Step, Ownable) {
-    super._transferOwnership(_newOwner);
+  /// @inheritdoc IConnextVestingWallet
+  function release() public virtual {
```
Comment (0xGorilla) [inline solidity/contracts/ConnextVestingWallet.sol:74]: no need for virtual

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+   * @return _timedelta The timedelta of the cliff duration
+   */
+  function VESTING_CLIFF_DURATION() external view returns (uint64 _timedelta);
+
+  /**
+   * NOTE:  The equivalent vesting schedule has a 13 months duration, with a 1 month cliff,
+   *        offsetted to start from `Sept 5th 2024 - 1 month`: At Sept 5th 2024 the cliff
+   *        is triggered unlocking 1/13 of the tokens, and then 1/13 of the tokens will
+   *        be linearly unlocked every month after that.
+   */
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:69]: mv to a significant place

File: solidity/contracts/ConnextVestingWallet.sol [hunk 3]
```diff
@@ -16,184 +15,91 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
-   * @notice This contract is only meant to unlock NEXT tokens
-   */
-  function vestedAmount(address _token, uint64 _timestamp) public view virtual override returns (uint256 _amount) {
-    if (_token != NEXT_TOKEN) return 0;
+  /// @inheritdoc IConnextVestingWallet
+  uint256 public released;
 
-    return _vestingSchedule(TOTAL_AMOUNT, _timestamp);
+  /// @param _owner  The address of the beneficiary
+  /// @param _totalAmount  The total amount of tokens to be unlocked
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:56]: multi line comment 🙏

File: solidity/contracts/ConnextVestingWallet.sol [hunk 4]
```diff
@@ -16,184 +15,91 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
-   */
+  /// @inheritdoc IConnextVestingWallet
   address public constant NEXT_TOKEN = 0xFE67A4450907459c3e1FFf623aA927dD4e28c67a;
 
-  /**
-   * NOTE:  The equivalent vesting schedule has a 13 months duration, with a 1 month cliff,
-   *        offsetted to start from `Sept 5th 2024 - 1 month`: At Sept 5th 2024 the cliff
-   *        is triggered unlocking 1/13 of the tokens, and then 1/13 of the tokens will
-   *        be linearly unlocked every month after that.
-   */
```
Comment (0xGorilla) [inline solidity/contracts/ConnextVestingWallet.sol:56]: Move this on top of vestedAmount

File: solidity/test/integration/LlamaVesting.t.sol [hunk 5]
```diff
@@ -8,7 +8,7 @@ contract IntegrationLlamaVesting is IntegrationBase {
 
   function setUp() public override {
     super.setUp();
-    _vestingStartTime = _connextVestingWallet.start();
+    _vestingStartTime = _connextVestingWallet.VESTING_START_DATE() + _connextVestingWallet.VESTING_DURATION();
```
Comment (wei3erHase) [inline solidity/test/integration/LlamaVesting.t.sol:11]: delete!

File: solidity/test/integration/LlamaVesting.t.sol [hunk 6]
```diff
@@ -8,7 +8,7 @@ contract IntegrationLlamaVesting is IntegrationBase {
 
   function setUp() public override {
     super.setUp();
-    _vestingStartTime = _connextVestingWallet.start();
```
[the remark this answers is not in the document]
Comment (dristpunk) [reply solidity/test/integration/LlamaVesting.t.sol:11]: > why?

not used, deleted

File: solidity/contracts/ConnextVestingWallet.sol [hunk 7]
```diff
@@ -16,184 +15,101 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
   /**
-   * @inheritdoc VestingWallet
-   * @notice This contract is only meant to unlock NEXT tokens
+   * @param _owner The address of the beneficiary
+   * @param _totalAmount The total amount of tokens to be unlocked
    */
-  function vestedAmount(uint64) public view virtual override returns (uint256 _amount) {
-    return 0;
+  constructor(address _owner, uint256 _totalAmount) Ownable(_owner) {
+    CLIFF = VESTING_START_DATE + VESTING_CLIFF_DURATION;
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:60]: CLIFF is basically `VESTING_START_DATE + 1 year` no?

File: solidity/contracts/ConnextVestingWallet.sol [hunk 8]
```diff
@@ -16,184 +15,101 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
-   */
-  function releasable(address _token) public view virtual override returns (uint256 _amount) {
-    _amount = vestedAmount(_token, uint64(block.timestamp)) - released(_token);
-    uint256 _balance = IERC20(_token).balanceOf(address(this));
-    _amount = _balance < _amount ? _balance : _amount;
+  /// @inheritdoc IConnextVestingWallet
+  function vestedAmount(uint64 _timestamp) public view returns (uint256 _amount) {
+    if (_timestamp < CLIFF) {
+      return 0;
+    } else if (_timestamp >= VESTING_START_DATE + VESTING_DURATION) {
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:75]: why is it not using CLIFF?

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+interface IConnextVestingWallet {
+  /*///////////////////////////////////////////////////////////////
+                             EVENTS
+  //////////////////////////////////////////////////////////////*/
+
+  /**
+   * @notice NEXT tokens released
+   * @param _token  The address of the NEXT token
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:13]: *of the released ERC20 token

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+
+  /**
+   * @notice Permission denied
+   */
+  error NotAllowed();
+
+  /**
+   * @notice Zero address not allowed
+   */
+  error ZeroAddress();
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:30]: is this one being used?

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+   * @return _timedelta The timestamp of Sept 5th 2023
+   */
+  function SEPT_05_2023() external view returns (uint64 _timedelta);
+
+  /**
+   * @notice Vesting cliff duration
+   * @dev 1 month cliff
+   * @return _timedelta The timedelta of the cliff duration
+   */
+  function VESTING_CLIFF_DURATION() external view returns (uint64 _timedelta);
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:73]: not a timedelta, but a unix timestamp

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+   * @return _timedelta The timedelta of one year
+   */
+  function ONE_YEAR() external view returns (uint64 _timedelta);
+
+  /**
+   * @notice Sept 5th 2023 in seconds
+   * @return _timedelta The timestamp of Sept 5th 2023
+   */
+  function SEPT_05_2023() external view returns (uint64 _timedelta);
+
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:67]: not a timedelta, unix timestamp!

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+  /**
+   * @notice Vesting duration including one month of cliff
+   * @dev 13 months duration
+   * @return _timedelta The timedelta of the vesting duration
+   */
+  function VESTING_DURATION() external view returns (uint64 _timedelta);
+
+  /**
+   * @notice Vesting warmup time
+   * @dev 11 months offset
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:84]: i'm not sure if `@dev` comments are correct in the interface, because they're talking about the immutable variable that's on the contract, it doesn't make sense to assume in the interface that this will return `11 months offset`

i'd move the `@dev` comments to the contract

File: solidity/contracts/ConnextVestingWallet.sol (hunk 7 above)
Comment (dristpunk) [reply solidity/contracts/ConnextVestingWallet.sol:60]: > CLIFF is basically `VESTING_START_DATE + 1 year` no?

 `VESTING_START_DATE + 1 month`. It's used just once so we could replace it with `VESTING_START_DATE + VESTING_CLIFF_DURATION` in the code

File: solidity/interfaces/IConnextVestingWallet.sol (hunk 10 above)
Comment (dristpunk) [reply solidity/interfaces/IConnextVestingWallet.sol:30]: > is this one being used?

once, at the dust collection

File: solidity/contracts/ConnextVestingWallet.sol (hunk 7 above)
Comment (wei3erHase) [reply solidity/contracts/ConnextVestingWallet.sol:60]: please do 🙏

Comment (dristpunk) [reply solidity/contracts/ConnextVestingWallet.sol:60]: > VESTING_CLIFF

I'd remove it at all or moved its declaration to constants. Setting constants in constructor doesn't make sence

File: solidity/contracts/ConnextVestingWallet.sol [hunk 14]
```diff
@@ -16,183 +15,108 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant VESTING_START_DATE = NEXT_TOKEN_LAUNCH + VESTING_OFFSET; // Sept 5th 2024 - 1 month
 
-  /**
-   * @notice Vesting start date
-   * @dev Sept 5th 2024 - 1 month
-   */
-  uint64 public constant VESTING_START_DATE = NEXT_TOKEN_LAUNCH + VESTING_OFFSET;
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant VESTING_CLIFF = VESTING_START_DATE + VESTING_CLIFF_DURATION;
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:47]: missing `// Sept 5th 2024` comment

File: solidity/interfaces/IConnextVestingWallet.sol [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+  function SEPT_05_2023() external view returns (uint64 _timestamp);
+
+  /**
+   * @notice Vesting cliff duration
+   * @return _timedelta The timedelta of the cliff duration
+   */
+  function VESTING_CLIFF_DURATION() external view returns (uint64 _timedelta);
+
+  /**
+   * @notice Vesting duration including one month of cliff
```
Comment (wei3erHase) [inline solidity/interfaces/IConnextVestingWallet.sol:73]: *including cliff duration

File: solidity/contracts/ConnextVestingWallet.sol [hunk 16]
```diff
@@ -16,183 +15,108 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant NEXT_TOKEN_LAUNCH = SEPT_05_2023; // Equals to Sept 5th 2023
 
-  /**
-   * @notice NEXT token address
-   * @dev Mainnet address
-   */
-  address public constant NEXT_TOKEN = 0xFE67A4450907459c3e1FFf623aA927dD4e28c67a;
+  /// @inheritdoc IConnextVestingWallet
+  address public constant NEXT_TOKEN = 0xFE67A4450907459c3e1FFf623aA927dD4e28c67a; // Mainnet address
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:32]: * Mainnet NEXT token address

File: solidity/contracts/ConnextVestingWallet.sol [hunk 17]
```diff
@@ -16,183 +15,108 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
-   * @inheritdoc VestingWallet
-   * @notice This contract is only meant to unlock NEXT tokens
-   */
-  function vestedAmount(uint64) public view virtual override returns (uint256 _amount) {
-    return 0;
+  /// @inheritdoc IConnextVestingWallet
+  function vestedAmount(uint64 _timestamp) public view returns (uint256 _amount) {
+    if (_timestamp < VESTING_CLIFF) {
+      return 0;
+    } else if (_timestamp >= VESTING_START) {
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:77]: how come this is passing the tests?
```suggestion
    } else if (_timestamp >= VESTING_END) {
```

File: solidity/contracts/ConnextVestingWallet.sol [hunk 18]
```diff
@@ -26,28 +26,31 @@ contract ConnextVestingWallet is Ownable2Step, IConnextVestingWallet {
...
-  uint64 public constant VESTING_START_DATE = NEXT_TOKEN_LAUNCH + VESTING_OFFSET;
+  uint64 public constant VESTING_START_DATE = NEXT_TOKEN_LAUNCH + VESTING_OFFSET; // Sept 5th 2024 - 1 month
 
   /// @inheritdoc IConnextVestingWallet
-  uint256 public immutable TOTAL_AMOUNT;
+  uint64 public constant VESTING_CLIFF = VESTING_START_DATE + VESTING_CLIFF_DURATION;
 
   /// @inheritdoc IConnextVestingWallet
-  uint64 public immutable CLIFF;
+  uint64 public constant VESTING_START = VESTING_START_DATE + VESTING_DURATION;
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:50]: WRONG NAMING!!

File: solidity/contracts/ConnextVestingWallet.sol [hunk 19]
```diff
@@ -16,183 +15,108 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant VESTING_OFFSET = ONE_YEAR - ONE_MONTH; // 11 months offset
 
-  /**
-   * @notice Vesting warmup time
-   * @dev 11 months offset
-   */
-  uint64 public constant VESTING_OFFSET = ONE_YEAR - ONE_MONTH;
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant VESTING_START = NEXT_TOKEN_LAUNCH + VESTING_OFFSET; // Aug 5th 2024
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:44]: why the change pal?? it was ok `Sept 5th - 1mth`

File: solidity/contracts/ConnextVestingWallet.sol [hunk 20]
```diff
@@ -16,183 +15,108 @@ import {IVestingEscrowSimple} from 'interfaces/IVestingEscrowSimple.sol';
...
-   */
-  uint64 public constant VESTING_START_DATE = NEXT_TOKEN_LAUNCH + VESTING_OFFSET;
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant VESTING_CLIFF = VESTING_START + VESTING_CLIFF_DURATION; // Sept 5th 2024
 
-  /*///////////////////////////////////////////////////////////////
-                             STORAGE
-  //////////////////////////////////////////////////////////////*/
+  /// @inheritdoc IConnextVestingWallet
+  uint64 public constant VESTING_END = VESTING_START + VESTING_DURATION; // Sept 5th 2025
```
Comment (wei3erHase) [inline solidity/contracts/ConnextVestingWallet.sol:50]: i believe we should be talking about `UNLOCK` instead of `VEST`, @0xGorilla ?

File: solidity/test/utils/Constants.sol [hunk 21]
```diff
@@ -21,7 +21,7 @@ contract Constants {
   // Vesting info
   uint64 public constant AUG_01_2022 = 1_659_312_000; // vesting start date
   uint64 public constant SEP_05_2023 = 1_693_872_000; // launch date
-  uint64 public constant VESTING_DURATION = 365 days * 4;
+  uint64 public constant UNLOCK_DURATION = 365 days * 4;
```
Comment (wei3erHase) [inline solidity/test/utils/Constants.sol:24]: NOOOOOOO!O!O!!O!O!O!O! hahhaha
this is actually the vest that goes into llamapay, it's 4ys, is a VEST

**Review by wei3erHase — DISMISSED** [review]
LGTM! 👍

**Review by turtlemoji — CHANGES_REQUESTED** [review]
please change test pk from env to tests, and random would be better !

Also:
```sh
Encountered 1 failing test in solidity/test/integration/LlamaVesting.t.sol:IntegrationLlamaVesting
[FAIL. Reason: setup failed: environment variable "DEPLOYER_PRIVATE_KEY" not found] setUp() (gas: 0)
```

It seems it needs `DEPLOYER_PRIVATE_KEY` (the same you are setting on the action rn), but its not implemented on the `.env.example` so it fails locally.

File: solidity/test/integration/ConnextVestingWallet.t.sol [hunk 22]
```diff
@@ -70,27 +70,19 @@ contract UnitConnextVestingWallet is Test, Constants {
...
+    assertEq(_connextVestingWallet.vestedAmount(_connextTokenLaunch), 0);
+    assertEq(_connextVestingWallet.vestedAmount(_firstMilestoneTimestamp - 1), 0);
 
-    assertEq(_connextVestingWallet.vestedAmount(NEXT_TOKEN_ADDRESS, _firstMilestoneTimestamp), 1 ether);
+    assertEq(_connextVestingWallet.vestedAmount(_firstMilestoneTimestamp), 1 ether);
 
-    assertApproxEqAbs(
-      _connextVestingWallet.vestedAmount(NEXT_TOKEN_ADDRESS, _firstMilestoneTimestamp + MONTH), 2 ether, MAX_DELTA
-    );
+    assertApproxEqAbs(_connextVestingWallet.vestedAmount(_firstMilestoneTimestamp + MONTH), 2 ether, MAX_DELTA);
```
Comment (turtlemoji) [inline solidity/test/integration/ConnextVestingWallet.t.sol:78]: Why MAX_DELTA is so high? (1 ether)

Comment (dristpunk) [reply solidity/test/integration/ConnextVestingWallet.t.sol:78]: > Why MAX_DELTA is so high? (1 ether) I would use just `assertEq` on this case and all this file cases

mostly because we use approximate datatimes, like 30 days = 1 month, but 1 month also = 1/12 year, which is 365 days. So its many

Comment (turtlemoji) [reply solidity/test/integration/ConnextVestingWallet.t.sol:78]: But in this case I mean.
`assertApproxEqAbs(_connextVestingWallet.vestedAmount(_firstMilestoneTimestamp + MONTH), 2 ether, MAX_DELTA);`

You are comparing the vestedAmount, to 2 ether. With a delta of 1 ether. In this specific case it is wrong

**Review by 0xGorilla — APPROVED** [review]

**Review by gas1cent — APPROVED** [review]
